### ISSUE TITLE
"Can't build lexer" error on Python 3.11-3.12

### ISSUE THREAD

**Issue by BetaRays** [thomasnyman/arxivprep#1]
Running with Python 3.12 or Python 3.11 produces this error, Python 3.10 works properly.
```
ERROR: /tmp/tmp.Kfqpbr0125/arxivprep/striptexcomments:59: Invalid regular expression for rule 't_BOLPCT'. global flags not at the start of the expression at position 13
Traceback (most recent call last):
  File "/tmp/tmp.Kfqpbr0125/arxivprep/striptexcomments", line 208, in <module>
    main()
  File "/tmp/tmp.Kfqpbr0125/arxivprep/striptexcomments", line 205, in main
    print(strip_comments(source).encode(args.encoding))
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp.Kfqpbr0125/arxivprep/striptexcomments", line 190, in strip_comments
    lexer = ply.lex.lex()
            ^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/ply/lex.py", line 909, in lex
    raise SyntaxError("Can't build lexer")
SyntaxError: Can't build lexer
```

**Comment by BetaRays** [comment]
[The only 3.11 change in the `re` module](https://docs.python.org/3.12/library/re.html#regular-expression-syntax) is this one:
> (?aiLmsux)
>     (One or more letters from the set 'a', 'i', 'L', 'm', 's', 'u', 'x'.) The group matches the empty string; the letters set the corresponding flags for the entire regular expression:
> - [re.A](https://docs.python.org/3.12/library/re.html#re.A) (ASCII-only matching)
> - [re.I](https://docs.python.org/3.12/library/re.html#re.I) (ignore case)
> - [re.L](https://docs.python.org/3.12/library/re.html#re.L) (locale dependent)
> - [re.M](https://docs.python.org/3.12/library/re.html#re.M) (multi-line)
> - [re.S](https://docs.python.org/3.12/library/re.html#re.S) (dot matches all)
> - [re.U](https://docs.python.org/3.12/library/re.html#re.U) (Unicode matching)
> - [re.X](https://docs.python.org/3.12/library/re.html#re.X) (verbose)
> 
>     (The flags are described in [Module Contents](https://docs.python.org/3.12/library/re.html#contents-of-module-re).) This is useful if you wish to include the flags as part of the regular expression, instead of passing a flag argument to the [re.compile()](https://docs.python.org/3.12/library/re.html#re.compile) function. Flags should be used first in the expression string.
> 
>     Changed in version 3.11: This construction can only be used at the start of the expression.

This seems to cause the issue as this patch fixes this error (though I didn’t check for correctness otherwise yet).
```patch
diff --git a/striptexcomments b/striptexcomments
index 88fe7d8..91c0081 100755
--- a/striptexcomments
+++ b/striptexcomments
@@ -26,7 +26,7 @@
 # python striptexcomments input.tex > output.tex
 # python striptexcomments input.tex -e encoding > output.tex
 #
-import ply.lex, argparse, io
+import ply.lex, argparse, io, re

 def strip_comments(source):
     tokens = (
@@ -57,7 +57,7 @@ def strip_comments(source):

     # Comments from % at beginning of line to end of line
     def t_BOLPCT(t):
-        r"(?m)^\s*\%"
+        r"^\s*\%"
         # ply uses the re module internally. In the re module ^ is by default
         # interpreted only at the beginning of a string and NOT at beginning of
         # every line. The multi-line flag (?m) changes this behavior to make ^
@@ -187,7 +187,7 @@ def strip_comments(source):
     def t_ANY_error(t):
         t.lexer.skip(1)

-    lexer = ply.lex.lex()
+    lexer = ply.lex.lex(reflags=int(re.VERBOSE | re.MULTILINE))
     lexer.input(source)
     return u"".join([tok.value for tok in lexer])
```

**Comment by BetaRays** [comment]
> [The only 3.11 change in the `re` module](https://docs.python.org/3.12/library/re.html#regular-expression-syntax) is this one:
> 
> > (?aiLmsux)
> > (One or more letters from the set 'a', 'i', 'L', 'm', 's', 'u', 'x'.) The group matches the empty string; the letters set the corresponding flags for the entire regular expression:
> > 
> > * [re.A](https://docs.python.org/3.12/library/re.html#re.A) (ASCII-only matching)
> > * [re.I](https://docs.python.org/3.12/library/re.html#re.I) (ignore case)
> > * [re.L](https://docs.python.org/3.12/library/re.html#re.L) (locale dependent)
> > * [re.M](https://docs.python.org/3.12/library/re.html#re.M) (multi-line)
> > * [re.S](https://docs.python.org/3.12/library/re.html#re.S) (dot matches all)
> > * [re.U](https://docs.python.org/3.12/library/re.html#re.U) (Unicode matching)
> > * [re.X](https://docs.python.org/3.12/library/re.html#re.X) (verbose)
> >   (The flags are described in [Module Contents](https://docs.python.org/3.12/library/re.html#contents-of-module-re).) This is useful if you wish to include the flags as part of the regular expression, instead of passing a flag argument to the [re.compile()](https://docs.python.org/3.12/library/re.html#re.compile) function. Flags should be used first in the expression string.
> >   Changed in version 3.11: This construction can only be used at the start of the expression.
> 
> This seems to cause the issue as this patch fixes this error (though I didn’t check for correctness otherwise yet).
> 
> ```diff
> diff --git a/striptexcomments b/striptexcomments
> index 88fe7d8..91c0081 100755
> --- a/striptexcomments
> +++ b/striptexcomments
> @@ -26,7 +26,7 @@
>  # python striptexcomments input.tex > output.tex
>  # python striptexcomments input.tex -e encoding > output.tex
>  #
> -import ply.lex, argparse, io
> +import ply.lex, argparse, io, re
> 
>  def strip_comments(source):
>      tokens = (
> @@ -57,7 +57,7 @@ def strip_comments(source):
> 
>      # Comments from % at beginning of line to end of line
>      def t_BOLPCT(t):
> -        r"(?m)^\s*\%"
> +        r"^\s*\%"
>          # ply uses the re module internally. In the re module ^ is by default
>          # interpreted only at the beginning of a string and NOT at beginning of
>          # every line. The multi-line flag (?m) changes this behavior to make ^
> @@ -187,7 +187,7 @@ def strip_comments(source):
>      def t_ANY_error(t):
>          t.lexer.skip(1)
> 
> -    lexer = ply.lex.lex()
> +    lexer = ply.lex.lex(reflags=int(re.VERBOSE | re.MULTILINE))
>      lexer.input(source)
>      return u"".join([tok.value for tok in lexer])
> ```

The patch doesn’t seem to produce different tar files except for timestamps on my project, so that might be good.

**Comment by thomasnyman** [comment]
Closed by #3
